src/libxrpl/protocol/Permissions.cpp

```diff
@@ -131,6 +131,19 @@ Permission::getGranularTxType(GranularPermissionType const& gpType) const
     return std::nullopt;
 }
 
+std::optional<std::reference_wrapper<uint256 const>> const
+Permission::getTxFeature(TxType txType) const
+{
+    auto const txFeaturesIt = txFeatureMap_.find(txType);
```

Comment:
Why isn't this directly pulling from `TxFormats` and instead storing it separately?

Reply:
> Why isn't this directly pulling from `TxFormats` and instead storing it separately?

I'm just following on how it was implemented in #5650. However, I think the way it was done in #5650 makes sense, because `TxFormats` is really little more than a specialization of `KnownFormats`, and neither of them seem to have a way to shoehorn extra information beyond the layout info itself. Since a new map would have to be added either way, it seems roughly equivalent to do it here in `Permissions`. I think it's a bit more appropriate here, since the feature grants _permission_ for the transaction to run.